app/townsquare/models.py

```diff
@@ -375,13 +376,31 @@ def __str__(self):
 
 class Favorite(SuperModel):
     """Model for each favorite."""
+    
     user = models.ForeignKey(get_user_model(), on_delete=models.CASCADE, related_name='favorites')
     activity = models.ForeignKey('dashboard.Activity', on_delete=models.CASCADE)
     created = models.DateTimeField(auto_now=True)
 
     def __str__(self):
         return f"Favorite {self.activity.activity_type}:{self.activity_id} by {self.user}"
 
+class Pin(SuperModel):
+    """Model for each Pinned."""
+
+    user = models.ForeignKey(get_user_model(), on_delete=models.CASCADE, related_name='pins')
```

Comment:
thinking perhaps we want to cxhange how the pin's are associated. 

Unlike favourites which are a user one to many, pin's are a one to one for to the target activity feed, perhaps excluding any filters user based filters. we could take a simple hash of the target feed and that becomes our unique basis for the pin. that way we can pin for the main feed, any specific tribe, hackathon or the like with little to no overhead.

Reply:
@developerfred could you take a look at @androolloyd comments?

Reply:
Thanks @PixelantDesign and @androolloyd I will change this model.

Reply:
Fred ping me on chat and we can discuss the best way to create a unique key for each pin / activity feed.

Reply:
Thanks @androolloyd I will do it now.